lib/registries/addon/modifiers/with-branding.ts

```diff
@@ -0,0 +1,24 @@
+import Modifier from 'ember-oo-modifiers';
+
+import Brand from 'registries/services/brand';
+
+import { inject as service } from '@ember/service';
+
+class WithBrandingModifier extends Modifier {
+    @service brand!: Brand;
```

Comment:
Actually this brings up a good question of where will this modifier be applied? Is it meant to be applied on the element to be branded? Or will it be applied on some parent container, like `.ember-application` or something other container element?

Reply:
From our previous discussions, I believe we were going to have the `with-branding` modifier applied to some parent element, and then any components nested within that would have the brand styles applied. Is this correct?

Reply:
In response to passing the brand as an argument, will there be different brands being passed into the modifier at different times?

I thought that injecting the brand into the modifier would be cleaner and less code than injecting the brand service into every controller/component that uses the `with-branding`  modifier (in its template) and then passing it into the modifier.

Reply:
yes, the `with-branding` modifier should be applied to some parent element/container, and any branded components nested underneath will use the given brand.

i'd prefer to avoid having modifiers depend on global state, to the extent possible --  passing the brand as an arg would be more testable in isolation, more explicit, and would allow us to do something like this, if we ever want:
```hbs
{{#each @listOfProviders as |provider|}}
  <div {{with-branding provider.brand}}>
    <!-- provider info card, with branding! -->
  </div>
{{/each}}
```

Reply:
Ah ok, in that case it makes sense to pass the brand in. Will fix! 👍

Reply:
so i guess i want us to use the service to hold a global brand for now, to avoid refactoring irrelevant stuff, but build the modifier itself so it doesn't assume global state and won't make it *harder* to eventually remove that global state

trying to follow a few of those threads to concrete conclusions, i think that'd mean:
- load and apply the `osf` brand in the application route `app/application/*` (e.g. `{{with-branding this.osfBrand}}`)
- on the explicitly branded registries pages, load the provider's brand and apply in the template (e.g. `{{with-branding @model.brand}}`)
- i think the above is the goal-on-the-horizon -- branding is ideally/eventually entirely controlled by the modifier -- but right now there's more bookkeeping to do:
  - in explicitly branded routes, set the brand on the service in `afterModel`, then reset to `null` in `deactivate`
  - use the service to brand the registries navbar -- `null` brand falls back to the existing OSF logo with special dropdown